File: psana_examples/tags/V00-01-01/src/DumpPrinceton.cpp
//--------------------------------------------------------------------------
// File and Version Information:
// 	$Id$
//
// Description:
//	Class DumpPrinceton...
//
// Author List:
//      Andrei Salnikov
//
//------------------------------------------------------------------------

//-----------------------
// This Class's Header --
//-----------------------
#include "psana_examples/DumpPrinceton.h"

//-----------------
// C/C++ Headers --
//-----------------

//-------------------------------
// Collaborating Class Headers --
//-------------------------------
#include "MsgLogger/MsgLogger.h"
#include "psddl_psana/princeton.ddl.h"

//-----------------------------------------------------------------------
// Local Macros, Typedefs, Structures, Unions and Forward Declarations --
//-----------------------------------------------------------------------

using namespace psana_examples;
PSANA_MODULE_FACTORY(DumpPrinceton)

//		----------------------------------------
// 		-- Public Function Member Definitions --
//		----------------------------------------

namespace psana_examples {

//----------------
// Constructors --
//----------------
DumpPrinceton::DumpPrinceton (const std::string& name)
  : Module(name)
{
  m_src = configStr("source", "DetInfo(:Princeton)");
}

//--------------
// Destructor --
//--------------
DumpPrinceton::~DumpPrinceton ()
{
}

// Method which is called at the beginning of the calibration cycle
void 
DumpPrinceton::beginCalibCycle(Event& evt, Env& env)
{
  MsgLog(name(), trace, "in beginCalibCycle()");

  shared_ptr<Psana::Princeton::ConfigV1> config1 = env.configStore().get(m_src);
  if (config1.get()) {
    
    WithMsgLog(name(), info, str) {
      str << "Princeton::ConfigV1:";
      str << "\n  width = " << config1->width();
      str << "\n  height = " << config1->height();
      str << "\n  orgX = " << config1->orgX();
      str << "\n  orgY = " << config1->orgY();
      str << "\n  binX = " << config1->binX();
      str << "\n  binY = " << config1->binY();
      str << "\n  exposureTime = " << config1->exposureTime();
      str << "\n  coolingTemp = " << config1->coolingTemp();
      str << "\n  readoutSpeedIndex = " << config1->readoutSpeedIndex();
      str << "\n  readoutEventCode = " << config1->readoutEventCode();
      str << "\n  delayMode = " << config1->delayMode();
      str << "\n  frameSize = " << config1->frameSize();
      str << "\n  numPixels = " << config1->numPixels();
    }
    
  }
}

// Method which is called with event data
void 
DumpPrinceton::event(Event& evt, Env& env)
{
  shared_ptr<Psana::Princeton::FrameV1> frame = evt.get(m_src);
  if (frame.get()) {
    WithMsgLog(name(), info, str) {
      str << "Princeton::FrameV1:";
      str << "\n  readoutTime = " << frame->readoutTime();

      const uint16_t* data = frame->data();
      str << "\n  data =";
      for (int i = 0; i < 10; ++ i) {
        str << " " << data[i];
      }
      str << " ...";
    }
  }
}
  
} // namespace psana_examples

File: pyana_examples/tags/V00-00-25/src/dump_princeton.py
#--------------------------------------------------------------------------
# File and Version Information:
#  $Id$
#
# Description:
#  Pyana user analysis module dump_princeton...
#
#------------------------------------------------------------------------

"""Example module for accessing SharedIpimb data.

This software was developed for the LCLS project.  If you use all or 
part of it, please give an appropriate acknowledgment.

@version $Id$

@author Andy Salnikov
"""

#------------------------------
#  Module's version from SVN --
#------------------------------
__version__ = "$Revision$"
# $Source$

#--------------------------------
#  Imports of standard modules --
#--------------------------------
import sys
import logging

#-----------------------------
# Imports for other modules --
#-----------------------------
from pypdsdata import xtc

#----------------------------------
# Local non-exported definitions --
#----------------------------------

# local definitions usually start with _

#---------------------
#  Class definition --
#---------------------
class dump_princeton (object) :
    """Class whose instance will be used as a user analysis module. """

    #----------------
    #  Constructor --
    #----------------
    def __init__ ( self, source="" ) :
        """Class constructor takes the name of the data source.

        @param source   data source
        """
        
        self.m_src = source

    #-------------------
    #  Public methods --
    #-------------------
    def beginjob( self, evt, env ) :
        
        config = env.getPrincetonConfig(self.m_src)
        if config:
        
            print "%s: %s" % (config.__class__.__name__, self.m_src)
            
            print "  width =", config.width();
            print "  height =", config.height();
            print "  orgX =", config.orgX();
            print "  orgY =", config.orgY();
            print "  binX =", config.binX();
            print "  binY =", config.binY();
            print "  exposureTime =", config.exposureTime();
            print "  coolingTemp =", config.coolingTemp();
            print "  readoutSpeedIndex =", config.readoutSpeedIndex();
            print "  readoutEventCode =", config.readoutEventCode();
            print "  delayMode =", config.delayMode();
            print "  frameSize =", config.frameSize();

    def event( self, evt, env ) :
        """This method is called for every L1Accept transition.

        @param evt    event data object
        @param env    environment object
        """

        frame = evt.get(xtc.TypeId.Type.Id_PrincetonFrame, self.m_src)
        if frame:
            print "%s: %s" % (frame.__class__.__name__, self.m_src)
            
            print "  shotIdStart =", frame.shotIdStart()
            print "  readoutTime =", frame.readoutTime()
            
            img = frame.data()
            print "  data.shape =", img.shape
            print "  data =", img


    def endjob( self, evt, env ) :
        
        pass

